pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -412,6 +488,23 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 			},
 		},
 	}
+
+	if len(svcHostnameMap.Map) > 0 {
```

Comment:
Do we want to get the ball rolling on a ambient_multinetwork_enabled feature flag?

The changes here aren't too concerning (though might be nice just in case) but probably over time we will accumulate more and more changes that we may want to gate while development is ongoing?

Reply:
Yeah good idea. I'll get that added